tt_metal/impl/dispatch/command_queue_interface.hpp

```diff
@@ -64,8 +65,12 @@ struct dispatch_constants {
         return *inst;
     }
 
+    using prefetch_q_entry_type = uint16_t;
+
     static constexpr uint8_t MAX_NUM_HW_CQS = 2;
-    typedef uint16_t prefetch_q_entry_type;
+    static constexpr uint32_t DISPATCH_MESSAGE_ENTRIES = 16;
```

Comment:
Is this the max number of sub devices? It seems like it's used across the stack to represent that. It may be clearer to have another `MAX_NUM_SUBDEVICES = DISPATCH_MESSAGE_ENTRIES`.

Reply:
+1 I was looking at device changes and was wondering what `DISPATCH_MESSAGE_ENTRIES` was supposed to be.

Reply:
I've added a new static constexpr in Device, `static constexpr uint32_t MAX_NUM_SUB_DEVICES = dispatch_constants::DISPATCH_MESSAGE_ENTRIES;`, and updated the relevant use cases in device files to use this instead of `DISPATCH_MESSAGE_ENTRIES`. `DISPATCH_MESSAGE_ENTRIES` is still used where appropriate. Let me know if you think it should be switched elsewhere.